specification/security/resource-manager/Microsoft.Security/preview/2023-12-01-preview/securityContacts.json

```diff
@@ -0,0 +1,501 @@
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Deletes a security contact data": {
+            "$ref": "./examples/SecurityContacts/DeleteSecurityContact_example.json"
+          }
+        },
+        "tags": [
+          "Security Contacts"
+        ],
+        "description": "Delete security contact configurations for the subscription",
+        "operationId": "SecurityContacts_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/SecurityContactName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Security contact deleted"
+          },
+          "204": {
+            "description": "Security contact was not found"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SecurityContactList": {
+      "type": "object",
+      "description": "List of security contacts response",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "description": "List of security contacts",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SecurityContact"
+          },
+          "nextLink": {
+            "readOnly": true,
+            "type": "string",
+            "description": "The URI to fetch the next page."
+          }
+        }
+      }
+    },
+    "SecurityContact": {
+      "type": "object",
+      "description": "Contact details and configurations for notifications coming from Microsoft Defender for Cloud.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Security contact data",
+          "$ref": "#/definitions/SecurityContactProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "SecurityContactProperties": {
+      "type": "object",
+      "description": "Describes security contact properties",
+      "properties": {
+        "emails": {
+          "type": "string",
+          "description": "List of email addresses which will get notifications from Microsoft Defender for Cloud by the configurations defined in this security contact."
+        },
+        "phone": {
+          "type": "string",
+          "description": "The security contact's phone number"
+        },
+        "isEnabled": {
+          "type": "boolean",
```

Comment:
ARM recommends enums over booleans for future proof APIs.

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.
A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].

Reply:
This is a newly exposed security contact property, but it is already present in our backend and databases, so changing it would be difficult. However, in this particular case, I do not see any additional values that we may need. As I mentioned, our backend uses this property for all old versions, but it is not exposed.